### PR TITLE
Migrate to bench common

### PR DESCRIPTION
**Pull request by ttousai**

### CODE REVIEW

File: check/test.go [hunk 1]
```diff
@@ -1,194 +0,0 @@
...
-		var flagVal string
-		isset := match
-
-		if isset && t.Compare.Op != "" {
-			// Expects flags in the form;
-			// --flag=somevalue
-			// --flag
-			// somevalue
-			//pttn := `(` + t.Flag + `)(=)*([^\s,]*) *`
-			pttn := `(` + t.Flag + `)(=)*([^\s]*) *`
```
Comment (lizrice) [inline check/test.go]: Need to check why we removed this comma in the regexp, as the version in bench_common still has the version with the comma

File: check/controls_test.go [hunk 2]
```diff
@@ -1,41 +0,0 @@
...
-	"io/ioutil"
-	"testing"
-
-	yaml "gopkg.in/yaml.v2"
-)
-
-const cfgDir = "../cfg/"
-
-// validate that the files we're shipping are valid YAML
-func TestYamlFiles(t *testing.T) {
```
Comment (lizrice) [inline check/controls_test.go]: Let's not lose this valuable test please (it runs in the automated build and prevents us from shipping a faulty test file)

File: check/test_test.go [hunk 3]
```diff
@@ -1,121 +0,0 @@
...
-	"strings"
-	"testing"
-)
-
-var (
-	in       []byte
-	controls *Controls
-)
-
-func init() {
```
Comment (lizrice) [inline check/test_test.go]: Also let's make sure we keep these tests, they are useful

File: check/data [hunk 4]
```diff
@@ -1,160 +0,0 @@
----
```
Comment (lizrice) [inline check/data]: See comment elsewhere, I want to keep the tests that use this data

File: check/controls_test.go (hunk 2 above)
Comment (ttousai) [reply check/controls_test.go]: @lizrice the plan is to move all the test to bench-common. But I see your point about testing test files. I'll add the TestYamlFiles back in.

File: check/test_test.go (hunk 3 above)
Comment (ttousai) [reply check/test_test.go]: See previous comments.

File: check/data (hunk 4 above)
Comment (ttousai) [reply check/data]: See previous comment on test files.

**Comment by ttousai** [comment]
@lizrice I have added tests for the YAML definition files like I mentioned in my previous comments.

File: check/controls_test.go (hunk 2 above)
Comment (lizrice) [reply check/controls_test.go]: Thanks

File: check/test_test.go (hunk 3 above)
Comment (lizrice) [reply check/test_test.go]: I see we have something in bench_common called TestTestExecute, but I think we are losing the set of actual test cases below, for specific kube-bench tests. Don't we need those?

File: check/data (hunk 4 above)
Comment (lizrice) [reply check/data]: There are some new tests in bench_common that might cover at least some of this, but I suspect we are losing some useful tests here?

**Comment by ttousai** [comment]
@lizrice I have pushed the tests which are at risk of getting lost to bench-common.

**Comment by lizrice** [comment]
@ttousai I think we should close this PR as this set of changes is pretty out of date now. You OK with that?

**Comment by ttousai** [comment]
@lizrice please go ahead.

**Pull request closed without merging**